src/Omega_h_mesh.hpp

```diff
@@ -125,26 +125,50 @@ class Mesh {
   Graph ask_star(Int dim);
   Graph ask_dual();
 
-  Adj ask_revClass (Int edim);
+/* ask_revClass (Int edim, LOs g_ids): takes input of entity dimension
+ * 'edim', and an 1d array of model entity IDs on local part to return
```

Comment:
I assume it is valid to provide any model id on any part.  If there are no mesh entities classified on a given model id then no entities are returned for it.  If that is correct, I'd suggest describing it this way instead of saying that the model entities need to be 'on the local part'.

Reply:
"f there are no mesh entities classified on a given model id then no entities are returned for it. " is partly correct. The statement is incorrect only if the input model entity id is greater than the max class id on local part, there will be a exit failure from the ask_rc function. Doing an all to all communication to get a global max model id and then adding extra places in the offset array for all procs does not sound like a good idea to me, as it goes against the distributed mesh concept. I assume that the mesh will be looked at in paraview to find specific model entity ids so keeping track of which model ent is on which process should be fairly easy. let me know for comments

Reply:
A model-based simulation workflow only requires the user to have knowledge of the geometric model (or possibly an abstract representation of it in some cases) to define their problem.  These reverse classifcation and boundary field APIs are needed to support that workflow.  I think Figure 3 of this paper captures that concept: https://www.scorec.rpi.edu/REPORTS/1996-25.pdf

Can we just ignore a model id if it does not have any mesh entities classified on in a given part or is there something in the current implementation that prevents that from working?  If there is a problem, can you point me at the line of code?

I don't think a collective communication is necessary unless we want to have a 'debug' mode that checks the existence of the user provided model entity ids.  On that note, if we load a copy of the EGADs (or Simmetrix) model on each process no communication is needed to check if an id is valid.  That is a larger discussion though for another time about our general use of Omegah (i.e., we typically don't load a model and it is becoming a problem).

Reply:
please take a look at lines :https://github.com/joshia5/omega_h/blob/8c0132104d83627297bfbac280bdcbd910529bae/src/Omega_h_bfields.cpp#L62-L63

which calculate the degrees

Reply:
also, i am available to talk now over webex

Reply:
@cwsmith please look at this commit https://github.com/SCOREC/omega_h/pull/11/commits/a7453330d484794bd7b48f03764f70fb21c9102d, it creates the new csr based on num. input gEnts

Reply:
and passes the modified ctest

Reply:
resolved here https://github.com/SCOREC/omega_h/pull/11/commits/ee078482c91d0cc3707736b66026b2a87fb1a8fa, https://github.com/SCOREC/omega_h/pull/11/commits/a7453330d484794bd7b48f03764f70fb21c9102d